keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/_clusterDeployStrategies.kt

```diff
@@ -0,0 +1,34 @@
+package com.netflix.spinnaker.keel.orca
+
+import com.netflix.spinnaker.keel.api.ClusterDeployStrategy
+import com.netflix.spinnaker.keel.api.ClusterDeployStrategy.Companion.DEFAULT_WAIT_FOR_INSTANCES_UP
+import com.netflix.spinnaker.keel.api.Highlander
+import com.netflix.spinnaker.keel.api.RedBlack
+import java.time.Duration.ZERO
+
+/**
+ * Transforms [ClusterDeployStrategy] into the properties required for an Orca deploy stage.
+ */
+fun ClusterDeployStrategy.toOrcaJobProperties(): Map<String, Any?> =
+  when (this) {
+    is RedBlack -> mapOf(
+      "strategy" to "redblack",
+      "maxRemainingAsgs" to maxServerGroups,
+      "delayBeforeDisableSec" to delayBeforeDisable?.seconds,
+      "delayBeforeScaleDownSec" to delayBeforeScaleDown?.seconds,
+      "scaleDown" to resizePreviousToZero,
+      "rollback" to mapOf("onFailure" to rollbackOnFailure),
+      "stageTimeoutMs" to (
+        (waitForInstancesUp
+          ?: DEFAULT_WAIT_FOR_INSTANCES_UP) +
+          (delayBeforeDisable ?: ZERO) +
+          (delayBeforeScaleDown ?: ZERO)
+        ).toMillis(),
+      "interestingHealthProviderNames" to if (considerOnlyInstanceHealth) listOf("Amazon") else null
```

Comment:
I think the health provider names should be passed into the function as a parameter because this is in keel-orca and not in keel-ec2-*. That way it's flexible and keeps the "amazon" in the amazon place